src/lm/tools/activePullRequestTool.ts

```diff
@@ -21,18 +29,171 @@ export class ActivePullRequestTool implements vscode.LanguageModelTool<FetchIssu
+			return !latest || new Date(run.created_at) > new Date(latest.created_at)
+				? run
+				: latest;
+		}, null);
+
+		if (!lastRun) {
+			return '';
+		}
+
+		const logsZip = await fetch(lastRun.logs_url, {
+			headers: {
+				Authorization: `Bearer ${token}`,
+				Accept: 'application/json',
+			},
+		});
+		const logsText = await logsZip.arrayBuffer();
+		const copilotSteps: string[] = [];
+		const zip = await JSZip.loadAsync(logsText);
+		for (const fileName of Object.keys(zip.files)) {
+			const file = zip.files[fileName];
+			if (!file.dir && fileName.endsWith('Processing Request.txt')) {
+				const content = await file.async('string');
+				copilotSteps.push(...content.split('\n'));
+			}
+		}
+		// Summarize the Copilot agent's thinking process using the model
+		const messages = [
+			vscode.LanguageModelChatMessage.Assistant('You are an expert summarizer. The following logs show the thinking process and performed actions of a GitHub Copilot agent that was in charge of working on the current pull request. Read the logs and always maintain the thinking process. You can remove information on the tool call results that you think are not necessary for building context.'),
+			vscode.LanguageModelChatMessage.User(`Copilot Agent Logs (JSON):\n${JSON.stringify(copilotSteps)}`)
+		];
+
+		let summaryText: string | undefined;
+		try {
+			const response = await model.sendRequest(messages, { justification: 'Summarizing Copilot agent logs for the active pull request.' }, cancellationToken);
+			summaryText = await (typeof response.text === 'string' ? response.text : (typeof response.text?.[Symbol.asyncIterator] === 'function' ? (async () => { let out = ''; for await (const chunk of response.text) { out += chunk; } return out; })() : ''));
+		} catch (e) {
+			summaryText = '';
+		}
+
+		return summaryText;
+	}
+
+	async fetchCodingAgentSession(
+		pullRequest: PullRequestModel,
+		model: vscode.LanguageModelChat,
+		cancellationToken: vscode.CancellationToken
+	): Promise<string | string[]> {
+		let authProvider: AuthProvider | undefined;
+		if (this.credentialStore.isAuthenticated(AuthProvider.githubEnterprise) && hasEnterpriseUri()) {
+			authProvider = AuthProvider.githubEnterprise;
+		} else if (this.credentialStore.isAuthenticated(AuthProvider.github)) {
+			authProvider = AuthProvider.github;
+		} else {
+			return [];
+		}
+		const github = this.credentialStore.getHub(authProvider);
+		const { token } = await github?.octokit.api.auth() as { token: string };
+		let copilotSteps: string | string[] = [];
+		try {
+			const sessionsResponse = await fetch(`https://api.githubcopilot.com/agents/sessions/resource/pull/${pullRequest.id}`, {
+				headers: {
+					Authorization: `Bearer ${token}`,
+					Accept: 'application/json',
+				},
+			});
+			if (!sessionsResponse.ok) {
+				throw new Error(`Failed to fetch sessions: ${sessionsResponse.statusText}`);
+			}
+			const sessions = await sessionsResponse.json();
```

Comment:
It would be great to have a type for this too!

Reply:
Added the return object type